lib/redux/actions/async.js

```diff
@@ -422,105 +422,134 @@ export function doVersionCheck() {
+          dispatch(syncActions.updateProfileSuccess());
+        }
+        return dispatch(syncActions.setPage(pages.MAIN));
       });
-    });
-    const { localStore } = services;
-    dispatch(syncActions.retrieveUsersTargetsFromStorage());
-    const devicesInStorage = localStore.getItem('devices') || {};
-    dispatch(syncActions.putUsersTargetsInStorage());
-    localStore.setItem(
-      'devices',
-      Object.assign({}, devicesInStorage, usersWithTargets)
-    );
-
-    if (!_.isEmpty(targetTimezones[uploadTargetUser]) &&
-      !_.isEmpty(targetDevices[uploadTargetUser])) {
-      dispatch(syncActions.setPage(pages.MAIN));
     }
-
-    const uploadsByUser = actionUtils.getDeviceTargetsByUser(usersWithTargets);
-
-    dispatch(syncActions.setUploads(uploadsByUser));
   };
 }
 
 export function retrieveTargetsFromStorage() {
   return (dispatch, getState) => {
     const { devices, uploadTargetUser } = getState();
     const { api, localStore } = services;
+    let fromLocalStore = false;
+
     dispatch(syncActions.retrieveUsersTargetsFromStorage());
-    const targets = localStore.getItem('devices');
-    if (targets === null) {
-      return dispatch(syncActions.setPage(pages.SETTINGS));
-    }
-    else {
+    let targets = localStore.getItem('devices');
+    if (targets !== null) {
+      fromLocalStore = true;
+      // wipe out the deprecated 'devices' localStore key
+      localStore.removeItem('devices');
       const uploadsByUser = actionUtils.getDeviceTargetsByUser(targets);
-
       dispatch(syncActions.setUploads(uploadsByUser));
+      dispatch(syncActions.setUsersTargets(targets));
     }
-    dispatch(syncActions.setUsersTargets(targets));
 
-    if (targets[uploadTargetUser] != null) {
-      const userTargets = targets[uploadTargetUser];
-      const targetDeviceKeys = _.pluck(userTargets, 'key');
+    const { targetDevices, targetTimezones } = getState();
+
+    // redirect based on having a supported device
+    if (!_.isEmpty(_.get(targetDevices, uploadTargetUser))) {
+      let usersWithTargets = {};
+      _.forOwn(targetDevices, (devicesArray, userId) => {
+        usersWithTargets[userId] = _.map(devicesArray, (deviceKey) => {
+          return {key: deviceKey};
+        });
+      });
+      _.forOwn(targetTimezones, (timezoneName, userId) => {
+        usersWithTargets[userId] = _.map(usersWithTargets[userId], (target) => {
+          if (timezoneName != null) {
+            target.timezone = timezoneName;
+          }
+          return target;
+        });
+      });
+      const targetDeviceKeys = targetDevices[uploadTargetUser];
       const supportedDeviceKeys = Object.keys(devices);
       const atLeastOneDeviceSupportedOnSystem = _.some(targetDeviceKeys, (key) => {
         return _.includes(supportedDeviceKeys, key);
       });
-      let timezones = [];
-      _.each(userTargets, (target) => {
-        if (target.timezone) {
-          timezones.push(target.timezone);
+
+      if(!fromLocalStore){
```

Comment:
More of a note for the future: I will be strongly in favor of linting spaces around `if (condition) {` in any new set-ups we have for visualization code, etc. The condition needs to breathe!